### PR TITLE
Apply MVC model

### PR DESCRIPTION
**Pull request by thanhtrantruong47**
Apply MVC model for practice create function render cart to UI, function login

### CODE REVIEW

File: practice-javascript/package.json [hunk 1]
```diff
@@ -5,6 +5,7 @@
   "scripts": {
     "start": "parcel src/index.html",
     "build": "parcel build ./src/index.html  --dist-dir ./dist"
+
```
Comment (vuphan-agilityio) [inline practice-javascript/package.json]: @thanhtrantruong47 remove empty line

File: practice-javascript/.gitignore [hunk 2]
```diff
@@ -3,5 +3,4 @@ node_modules
 dist
 dist-ssr
 .parcel-cache
-package-lock.json
-js
\ No newline at end of file
+package-lock.json
```
Comment (sinhnguyen-agilityio) [inline practice-javascript/.gitignore]: No need to ignore pack-lock.json

File: practice-javascript/src/js/controllers/signupController.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  e.preventDefault();
+  if (
+    username.value == "" ||
+    password.value == "" ||
+    fname.value == "" ||
+    lname.value == "" ||
+    avatar.value == "" ||
+    phonenumber.value == ""
+  ) {
+    alert("not empty");
```
Comment (sinhnguyen-agilityio) [inline practice-javascript/src/js/controllers/signupController.js]: What field can't be empty? If you show the alert like this, user don't know how to fill the right value

File: practice-javascript/src/js/controllers/signupController.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  } else {
+    const user = {
+      email: username.value,
+      password: password.value,
```
Comment (sinhnguyen-agilityio) [inline practice-javascript/src/js/controllers/signupController.js]: Can you add the field to re-enter the password?

File: practice-javascript/src/js/constants/url.js [hunk 5]
```diff
@@ -0,0 +1,6 @@
+export const API =  {
+  BASE_URL : "http://localhost:3000",
+  USER : "/user",
+  CART: "/cart"
```
Comment (vuphan-agilityio) [inline practice-javascript/src/js/constants/url.js]: @thanhtrantruong47 Use plural for path: `/users` and `/carts`

File: practice-javascript/src/js/controllers/signupController.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+      first_name: fname.value,
+      last_name: lname.value,
+      avatar_url: avatar.value,
+      phone_number: phonenumber.value,
+      role: "user",
+    };
+    fetch(api , {
```
Comment (sinhnguyen-agilityio) [inline practice-javascript/src/js/controllers/signupController.js]: What will happen if the signup process fails?

File: practice-javascript/src/js/models/userModel.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
+class UserModel {
+  constructor(id,emai,password,first_name,last_name,avatar_url,phone_number,role){
```
Comment (vuphan-agilityio) [inline practice-javascript/src/js/models/userModel.js]: @thanhtrantruong47 Follow at least one convention, you are missing a space after `,`

File: practice-javascript/src/js/services/connect.js [hunk 8]
```diff
@@ -0,0 +1,2 @@
+// import { API } from "../constants/url";
```
Comment (vuphan-agilityio) [inline practice-javascript/src/js/services/connect.js]: @thanhtrantruong47 Separate a service file is a good approach, but please help put all CRUD APIs calls into this place.

File: practice-javascript/src/js/views/userView.js [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+    this.password = document.querySelector("#password");
+    this.btnLogin = document.querySelector(".login__btn");
+    this.valiEmail = document.querySelector(".email-validate");
+    this.valipassword = document.querySelector(".password-validate");
+  }
+
+  handleLogin(dataUser) {
+    this.btnLogin.addEventListener("click", (e) => {
+      e.preventDefault()
+    if (this.username.value == "" || this.password.value == "") {
```
Comment (sinhnguyen-agilityio) [inline practice-javascript/src/js/views/userView.js]: Reduce the nested if else condition please

File: practice-javascript/src/js/constants/url.js [hunk 10]
```diff
@@ -0,0 +1,5 @@
+export const API = {
+  BASE_URL: "http://localhost:3000",
+  USERS: "/user",
+  CARTS: "/cart",
```
Comment (vuphan-agilityio) [inline practice-javascript/src/js/constants/url.js]: @thanhtrantruong47 Please check carefully

File: practice-javascript/src/js/services/userService.js [hunk 11]
```diff
@@ -27,8 +26,8 @@ export default class UserService {
     if (!res.ok) {
```
Comment (sinhnguyen-agilityio) [inline practice-javascript/src/js/services/userService.js]: Consider to user try catch for handle Async Await error

File: practice-javascript/src/js/views/userView.js [hunk 12]
```diff
@@ -53,21 +59,76 @@ class UserView {
...
+        return emailPattern.test(email);
+      }
+
+      function validatePhone(phone) {
+        var phoneRegex = /^0\d{9}$/;
+        return phoneRegex.test(phone);
+      }
+
+      try {
+        const isVali = true;
```
Comment (sinhnguyen-agilityio) [inline practice-javascript/src/js/views/userView.js]: isValid?

File: practice-javascript/src/js/services/cartService.js [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+  static deleteCart = async (id) => {
+    try {
+      const res = await fetch(`${API.BASE_URL}${API.CART}/${id}`, {
+        method: "DELETE",
+        headers: {
+          "Content-Type": "Application-json",
+        },
+      });
+    } catch (error) {
+      throw new Error("Failed to get cart.");
```
Comment (sinhnguyen-agilityio) [inline practice-javascript/src/js/services/cartService.js]: Failed to delete the cart?

**Pull request merged**